GFX tweaks & Address details (modal)

AllienWorks opened the pull request:
> WORK IN PROGRESS, don't merge

## Done

### Overview

- UI/UX unification (closes **`PD-390`**, **`PD-391`** & **`PD-392`**)
- tweaks to Staking widgets

### Send

- UI/UX unification (typography, sticky layout – [preview](https://user-images.githubusercontent.com/1965795/47515093-3917a880-d882-11e8-9420-b38833615019.png))
- `Advanced options` button moved to tab bar, so it's still visible and available (even when scrolled down)

### Help

- added links to forum, exchanges ([preview](https://user-images.githubusercontent.com/1965795/47515086-34eb8b00-d882-11e8-8e75-a974e9cd0b96.png))

### Receive & Address book

- redesigned list of addresses ([preview](https://user-images.githubusercontent.com/1965795/47996270-42153f00-e0f8-11e8-9851-ade5e39f4e88.png))
- Address book: redesigned "save new address" modal ([preview](https://user-images.githubusercontent.com/1965795/47996288-59542c80-e0f8-11e8-8dc0-2dc8a60b9fd7.png)) and "delete address"
- closes **`PD-414`**

### Address details

- modal updated (`qr-code-modal` – [preview](https://user-images.githubusercontent.com/1965795/47515078-2ef5aa00-d882-11e8-9aa0-2e0651ed0fbb.png))
- this will replace `new-address-modal` & `add-address-label`
- closes **`PD-422`**, **`PD-404`**, **`PD-405`**

### Misc.

- made help descriptions a bit larger + bigger line spacing so better readability

Commit:
Overview: widget tweaks, styling unification

fixes PD-390, PD-391 & PD-392

Commit:
Send: GFX tweaks, styling unification

+ moved [Adv. options], sticky "Pay from" box

Commit:
Receive: GFX tweaks

Commit:
Help: added forums & exchanges

Commit:
Address modal: layout basics

Commit:
Address modal: layout & styling

vikas-cis committed:
created shared component and used it on receive page

vikas-cis committed:
Added implementation on qr-code modal for edit address and update after edition also added verify modal action

vikas-cis committed:
fix tests and small issues

vikas-cis-zz commented:
@AllienWorks please have a look and let me know if there is anything that I missed or any changes required regarding css

Thanks

vikas-cis committed:
added the qr modal on receive page and done some tweaks there

AllienWorks commented:
@vikas-cis can you please look into these as well?

#### `qr-code-modal`

- [x] displaying addresses on Explorer doesn't work when on testnet (the testnet detection/condition in link isn't working)
- [x] switch `mat-dialog-titles` on top based on if you're looking at _MY address_ or _SOMEONE ELSE's address_

#### `detail-address`

- [x] address type: show address ID (e.g. "#43") only when displaying _MY_ address
- [x] switch help description also for _MY vs. SOMEONE ELSE'S_ address (it's marked TODO in comments)

I will take care of the styling in the meantime.. :)

Commit:
Address details/Receive: fixed styling

Commit:
Receive: used addresses redesign

vikas-cis-zz commented:
@AllienWorks yeah will check it out 

Thanks

Commit:
Address book: fixed layout + modal trigger

Commit:
Address list: redesign

HenkSwardt commented:
@AllienWorks @vikas-cis , Is this issue ready to be merged?

vikas-cis-zz commented:
@HenkSwardt no I need to do some changes over their for few things

Thanks

Commit:
Address book > New address: redesign

Commit:
Address book > Delete address: redesign

vikas-cis committed:
fix link for testnet, shows the description on the basis of owned addresses etc..

vikas-cis committed:
Merge branch 'gfx/tweaks' of github.com:AllienWorks/particl-desktop into gfx/tweaks

vikas-cis-zz commented:
@AllienWorks I have done all the requested changes, can you please have a look and let me know if all looks good :)

@HenkSwardt @zaSmilingIdiot Also it is ready for review from mine side :)

Thanks

AllienWorks commented:
Tested, all changes by @vikas-cis work as intented.
Overall I'm happy with all this = it's ready for code review @zaSmilingIdiot @HenkSwardt 

(Also sorry this PR got a bit bigger than it should be, I'll try to keep my feet on the ground more next time :)

HenkSwardt commented:
Thank you @AllienWorks and @vikas-cis 

Can I ask that one of you please remove the WIP from the label.

Thanks

vikas-cis committed:
trigger build

AllienWorks commented:
> Can I ask that one of you please remove the WIP from the label.

Can't do, insufficient permissions. That's on @zaSmilingIdiot :)

zaSmilingIdiot commented:
> > Can I ask that one of you please remove the WIP from the label.
> 
> Can't do, insufficient permissions. That's on @zaSmilingIdiot :)

Hehehe... Already done ;P

zaSmilingIdiot committed:
Updates new-address-modal verifyAddress method

The value of this.address can also be null (occurs when cancelling the modal without any input), hence the check for address validity was previously still occurring when the modal was cancelled. This change ensures that we trap all falsey values for this.address, ensuring that we only check validity if a non-empty string is input.

zaSmilingIdiot committed:
Update qr-code-modal.component.html

Removes invalid method handler for sign/verify of addresses. This needs to be implemented correctly. New task created for this.

zaSmilingIdiot committed:
Update detail-address.component.ts

Ensures editing of the label is disabled when navigating between public and private address tabs int he address book page.

Commit:
Cold staking (modals): fixed messages, redesign

Commit:
Merge branch 'market' of https://github.com/particl/particl-desktop into gfx/tweaks

zaSmilingIdiot committed:
Update qr-code-modal.component.html

Removes unnnecessary FIXME indicator.

zaSmilingIdiot committed:
fixes linting issues

Removing trailing whitespace
